File: 2015/02/02.py
# Read the input as list of strings
with open("02_input.txt", "r") as f:
    lines = f.read().splitlines()

wrap = 0
ribbon = 0
for line in lines:
    dims = list(map(int, line.split("x")))
    l, w, h = dims[0], dims[1], dims[2]
    wrap += 2 * l * w + 2 * w * h + 2 * h * l + min(l*w, w*h, h*l)
    dims.sort()
    smallest, second_smallest = dims[0], dims[1]
    ribbon += 2 * smallest + 2 * second_smallest + l * w * h

# Part One: Total square feet of wrapping paper
print("Part One:", wrap)

# Part Two: Total feet of ribbon
print("Part Two:", ribbon)

File: 2015/02/02.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strconv"
	"strings"
)

func main() {
	file, err := os.Open("2015/02/02-input.txt")
	if err != nil {
		panic(err)
	}
	scanner := bufio.NewScanner(file)
	var wrap, ribbon int
	for scanner.Scan() {
		// Parse dimensions
		dims := make([]int, 3)
		for i, dim := range strings.Split(scanner.Text(), "x") {
			dims[i], _ = strconv.Atoi(dim)
		}
		l, w, h := dims[0], dims[1], dims[2]

		// Calculate wrapping paper
		wrap += 2*l*w + 2*w*h + 2*h*l + min(min(l*w, w*h), h*l)

		// Calculate ribbon
		sort.Ints(dims)
		ribbon += 2*dims[0] + 2*dims[1] + l*w*h
	}
	defer must(file.Close())

	// Part One: Total square feet of wrapping paper
	fmt.Printf("Part One: %d\n", wrap)

	// Part Two: Total feet of ribbon
	fmt.Printf("Part Two: %d\n", ribbon)
}

func must(err error) {
	if err != nil {
		panic(err)
	}
}
